Documentation for downstream TLS client auth (#1645)
kroxylicious/kroxylicious: docs/modules/configuring/con-configuring-client-connections.adoc

> * Provide the TLS configuration using the tls properties in the virtual cluster's configuration to enable it to present the certificate to clients. 

Reviewer: ```suggestion
* Provide the TLS configuration using the `tls` properties in the virtual cluster's configuration to enable it to present the certificate to clients. 
```

Depending on your certificate format, apply one of the following examples.

> You may also configure the virtual cluster to use TLS client authentication.

Reviewer: Should we change the title of this section to "Securing client connections"

> You may also configure the virtual cluster to use TLS client authentication.

Author: Done.

# …
----
<1> Private key of the virtual cluster.
<2> Public certificate of the virtual cluster.
<3> (Optional) Password for the key.

> You may configure the virtual cluster to require that clients present a certificate to the virtual
> cluster. The virtual cluster verifies the client's certificate is signed by one of the CA certificates
> contained with a trust store.  If verification fails, the client's connection will be refused.

Reviewer: ```suggestion
You can configure the virtual cluster to require that clients present a certificate for authentication. 
The virtual cluster verifies that the client's certificate is signed by one of the CA certificates contained in a trust store.  
If verification fails, the client's connection is refused.
```

.Example to configure TLS client authentication using PKCS12 trust store
[source,yaml]
----
virtualClusters:
  demo:
    tls:
      key:
        # ...
      trust:
        storeFile: <path>/trust.p12 #1 <1>
        storePassword:
          passwordFile: <path>/trust.password # <2>
        storeType: PKCS12 # <3>
        trustOptions:
          clientAuth: REQUIRED # <4>
# …
----
<1> PKCS #12 store containing CA certificate(s) used to verify that the client's certificate is trusted.
<2> (Optional) Password to protect the PKCS #12 store.
<3> (Optional) Keystore type. If a keystore type is not specified, the default JKS (Java Keystore) type is used.

> <4> If set to the value `REQUIRED`, the client must present a valid certificate.  If set to `REQUESTED` the client
>     will be requested to present a certificate.  If a certificate is presented, it will be validated. If the client
>     chooses not to present a certificate the connection is still allowed.  The value `NONE` disables client authentication.

Reviewer: ```suggestion
<4> Client authentication mode. 
If set to `REQUIRED`, the client must present a valid certificate. 
If set to `REQUESTED`, the client is requested to present a certificate. If presented, the certificate is validated. If the client chooses not to present a certificate the connection is still allowed. 
If set to `NONE`, client authentication is disabled.
```

> NOTE: The client's identity established by TLS client authentication is not relayed to the target cluster. See

Reviewer: ```suggestion
NOTE: The client's identity, as established through TLS client authentication, is currently not relayed to the target cluster. 
```

> (https://github.com/kroxylicious/kroxylicious/issues/1637)[issue].

Reviewer: ```suggestion
For more information, see the https://github.com/kroxylicious/kroxylicious/issues/1637[related issue].
```

> 
> 
> 

Reviewer: ```suggestion
```
